gluten-substrait/src/main/scala/org/apache/gluten/expression/UnaryExpressionTransformer.scala

```diff
@@ -43,7 +43,10 @@ case class CastTransformer(substraitExprName: String, child: ExpressionTransform
   extends UnaryExpressionTransformer {
   override def doTransform(args: java.lang.Object): ExpressionNode = {
     val typeNode = ConverterUtils.getTypeNode(dataType, original.nullable)
-    ExpressionBuilder.makeCast(typeNode, child.doTransform(args), original.ansiEnabled)
+    ExpressionBuilder.makeCast(
+      typeNode,
+      child.doTransform(args),
+      original.evalMode == EvalMode.ANSI)
```

Comment:
This is added in spark 3.4. I think you only need to fix the behavior in spark 3.4 and after.

Reply:
https://github.com/apache/incubator-gluten/blob/main/shims/spark35/src/main/scala/org/apache/gluten/sql/shims/spark35/Spark35Shims.scala#L485

Reply:
Thank you! Used this shim function.